src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/MetadataReader.netstandard.cs

```diff
@@ -13,13 +14,15 @@ public sealed partial class MetadataReader
         internal AssemblyName GetAssemblyName(StringHandle nameHandle, Version version, StringHandle cultureHandle, BlobHandle publicKeyOrTokenHandle, AssemblyHashAlgorithm assemblyHashAlgorithm, AssemblyFlags flags)
         {
             string name = GetString(nameHandle);
-            // compat: normalize 'null' culture name to "" to match AssemblyName.GetAssemblyName()
+            // compat: normalize Nil culture name to "" to match original behavior of AssemblyName.GetAssemblyName()
             string cultureName = (!cultureHandle.IsNil) ? GetString(cultureHandle) : "";
             var hashAlgorithm = (Configuration.Assemblies.AssemblyHashAlgorithm)assemblyHashAlgorithm;
-            byte[]? publicKeyOrToken = !publicKeyOrTokenHandle.IsNil ? GetBlobBytes(publicKeyOrTokenHandle) : null;
+            // compat: native BaseAssemblySpec::Init guarantees that publicKeyOrToken is never null.
+            byte[]? publicKeyOrToken = !publicKeyOrTokenHandle.IsNil ? GetBlobBytes(publicKeyOrTokenHandle) : Array.Empty<byte>();
```

Comment:
It was not trivial to track this behavior to the native counterpart -

https://github.com/dotnet/runtime/blob/dd3cd5da4bc54f27e06c6dd080f2536ddbac6fd2/src/coreclr/vm/baseassemblyspec.inl#L328

In the native code we look at the length of the PKT and if it is 0, then we set the `m_cbPublicKeyOrToken` to `1`. 
Later we look at `m_cbPublicKeyOrToken` and seeing it is not `0` we would create an array of length `0`.

https://github.com/dotnet/runtime/blob/eb9dd67ce9fc64c468cf5c795fda7057310c22fc/src/coreclr/vm/assemblyspec.cpp#L458

Reply:
The second part was recently deleted when parsing was switched completely to managed. This is a link to an old version.